internal/processing/processors/v2alpha1/virtual_service_processor.go

```diff
@@ -0,0 +1,175 @@
+}
+
+// VirtualServiceCreator provides the creation of a Virtual Service using the configuration in the given APIRule.
+type VirtualServiceCreator interface {
+	Create(api *gatewayv2alpha1.APIRule) (*networkingv1beta1.VirtualService, error)
+}
+
+// EvaluateReconciliation evaluates the reconciliation of the Virtual Service for the given API Rule.
+func (r VirtualServiceProcessor) EvaluateReconciliation(ctx context.Context, client ctrlclient.Client) ([]*processing.ObjectChange, error) {
+	desired, err := r.getDesiredState(r.ApiRule)
+	if err != nil {
+		return make([]*processing.ObjectChange, 0), err
+	}
+
+	actual, err := r.getActualState(ctx, client, r.ApiRule)
+	if err != nil {
+		return make([]*processing.ObjectChange, 0), err
+	}
+
+	changes := r.getObjectChanges(desired, actual)
+
+	return []*processing.ObjectChange{changes}, nil
+}
+
+func (r VirtualServiceProcessor) getDesiredState(api *gatewayv2alpha1.APIRule) (*networkingv1beta1.VirtualService, error) {
+	return r.Creator.Create(api)
+}
+
+func (r VirtualServiceProcessor) getActualState(ctx context.Context, client ctrlclient.Client, api *gatewayv2alpha1.APIRule) (*networkingv1beta1.VirtualService, error) {
+	labels := getOwnerLabels(api)
+
+	var vsList networkingv1beta1.VirtualServiceList
+	if err := client.List(ctx, &vsList, ctrlclient.MatchingLabels(labels)); err != nil {
+		return nil, err
+	}
+
+	if len(vsList.Items) >= 1 {
+		return vsList.Items[0], nil
+	} else {
+		return nil, nil
+	}
+}
+
+func (r VirtualServiceProcessor) getObjectChanges(desired *networkingv1beta1.VirtualService, actual *networkingv1beta1.VirtualService) *processing.ObjectChange {
+	if actual != nil {
+		actual.Spec = *desired.Spec.DeepCopy()
+		return processing.NewObjectUpdateAction(actual)
+	} else {
+		return processing.NewObjectCreateAction(desired)
+	}
+}
+
+// The owner labels are still set to the old APIRule version.
+// Do not switch the owner labels to the new APIRule version unless absolutely necessary!
+// This has been done before, and it caused a lot of confusion and bugs.
+// If the change for some reason has to be done, please remove the version from the processing.OwnerLabel constant.
+func getOwnerLabels(api *gatewayv2alpha1.APIRule) map[string]string {
+	return map[string]string{
+		processing.OwnerLabel: fmt.Sprintf("%s.%s", api.ObjectMeta.Name, api.ObjectMeta.Namespace),
+	}
+}
+
+type virtualServiceCreator struct {
+	defaultDomainName string
+}
+
+// Create returns the Virtual Service using the configuration of the APIRule.
+func (r virtualServiceCreator) Create(api *gatewayv2alpha1.APIRule) (*networkingv1beta1.VirtualService, error) {
+	virtualServiceNamePrefix := fmt.Sprintf("%s-", api.ObjectMeta.Name)
+
+	vsSpecBuilder := builders.VirtualServiceSpec()
+	for _, host := range api.Spec.Hosts {
+		vsSpecBuilder.Host(default_domain.GetHostWithDomain(string(*host), r.defaultDomainName))
+	}
+
+	vsSpecBuilder.Gateway(*api.Spec.Gateway)
+	filteredRules := filterDuplicatePaths(api.Spec.Rules)
```

Comment:
Why do we need to filter at this point? We have a validation in place that will not allow rules with same path and methods. I think this can also lead to accidentally dropping relevant configuration if we support having multiple methods on the same path exposed different, e.g. expose "GET /users" with NoAuth and "POST /users" with JWT. This is probably only relevant if we add back the HTTP method configuration in the VS (`matchBuilder.MethodRegEx(rule.Methods...)`) .

I think a test like this is valid in that case:
```go
		Entry("test",
			newAPIRuleBuilder().
				WithHost("example-host.example.com").
				WithGateway("example-namespace/example-gateway").
				WithService("example-service", "example-namespace", 8080).
				WithRules(newRuleBuilder().WithMethods(http.MethodGet).WithPath("/").NoAuth().Build(),
					newRuleBuilder().WithMethods(http.MethodPut).WithPath("/").WithJWTAuthn("example.com", "https://jwks.example.com", nil, nil).Build()).Build(),
			[]verifier{
				func(vs *networkingv1beta1.VirtualService) {

					Expect(vs.Spec.Http[0].Match[0].Method.GetRegex()).To(Equal("^(GET|PUT)$"))
				},
			}, "create"),
```